<h4> 1.	The HOUSES dataset contains a variable SALE_PRICE.  Show how you would determine the median sale price.</h4>

In [3]:
import pandas as pd
import numpy as np 

In [4]:
df = pd.read_csv("https://raw.githubusercontent.com/ageron/handson-ml/master/datasets/housing/housing.csv")

In [5]:
df.head()

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0    -122.23     37.88                41.0        880.0           129.0   
1    -122.22     37.86                21.0       7099.0          1106.0   
2    -122.24     37.85                52.0       1467.0           190.0   
3    -122.25     37.85                52.0       1274.0           235.0   
4    -122.25     37.85                52.0       1627.0           280.0   

   population  households  median_income  median_house_value ocean_proximity  
0       322.0       126.0         8.3252            452600.0        NEAR BAY  
1      2401.0      1138.0         8.3014            358500.0        NEAR BAY  
2       496.0       177.0         7.2574            352100.0        NEAR BAY  
3       558.0       219.0         5.6431            341300.0        NEAR BAY  
4       565.0       259.0         3.8462            342200.0        NEAR BAY

In [6]:
df.describe()

longitude      latitude  housing_median_age   total_rooms  \
count  20640.000000  20640.000000        20640.000000  20640.000000   
mean    -119.569704     35.631861           28.639486   2635.763081   
std        2.003532      2.135952           12.585558   2181.615252   
min     -124.350000     32.540000            1.000000      2.000000   
25%     -121.800000     33.930000           18.000000   1447.750000   
50%     -118.490000     34.260000           29.000000   2127.000000   
75%     -118.010000     37.710000           37.000000   3148.000000   
max     -114.310000     41.950000           52.000000  39320.000000   

       total_bedrooms    population    households  median_income  \
count    20433.000000  20640.000000  20640.000000   20640.000000   
mean       537.870553   1425.476744    499.539680       3.870671   
std        421.385070   1132.462122    382.329753       1.899822   
min          1.000000      3.000000      1.000000       0.499900   
25%        296.000000    787.000000    280.000000       2.563400   
50%        435.000000   1166.000000    409.000000       3.534800   
75%        647.000000   1725.000000    605.000000       4.743250   
max       6445.000000  35682.000000   6082.000000      15.000100   

       median_house_value  
count        20640.000000  
mean        206855.816909  
std         115395.615874  
min          14999.000000  
25%         119600.000000  
50%         179700.000000  
75%         264725.000000  
max         500001.000000

In [7]:
np.median(df["median_house_value"])

179700.0

<h4>2.	The CUSTOMER dataset contains name and address information for every customer the company has and the unique identifier for the customer is CUSTOMER_NUMBER.  The PURCHASES dataset contains information on the purchases made by customers in the last three months and also contains the CUSTOMER_NUMBER variable.  Show how you would create a dataset containing all the customers that had not made any purchases in the last three months.</h4>

<h4>3.	The ACCOUNT table contains a list of accounts held by customers.  A customer may have more than one account of each type   The layout for the table is as follows:</h4>
<h5>
    <td> 
        <tr>CUSTOMER_NUMBER</tr>
        <tr>CUSTOMER_AGE</tr>
        <tr>ACCOUNT_NUMBER</tr>
        <tr>ACCOUNT TYPE</tr>
    </td>
    <td> 
        <tr>123</tr>
        <tr>27</tr>
        <tr>A987</tr>
        <tr>Home Loan</tr>
    </td>
    <td> 
        <tr>124</tr>
        <tr>37</tr>
        <tr>B6547</tr>
        <tr>Credit Card</tr>
    </td>
    <td> 
        <tr>125</tr>
        <tr>32</tr>
        <tr>B7531</tr>
        <tr>Credit Card</tr>
    </td>
    
   <br><br>
   Show how you determine the proportion of customers with a credit card for each of the following age ranges : 18 – 29, 30 – 44, 45 – 59, 60+.
  </h5>
  

In [8]:
df = pd.read_csv("account.csv")

In [9]:
df.columns

Index(['CUSTOMER_NUMBER', 'CUSTOMER_AGE', 'ACCOUNT_NUMBER', 'ACCOUNT TYPE',
       'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6', 'Unnamed: 7'],
      dtype='object')

In [11]:
df.head()

CUSTOMER_NUMBER  CUSTOMER_AGE ACCOUNT_NUMBER   ACCOUNT TYPE  Unnamed: 4  \
0     0b100001234            28    1000300A987       Car loan         NaN   
1     0b100001235            90    1000300A988  Personal Loan         NaN   
2     0b100001236            50    1000300A989      Home Loan         NaN   
3     0b100001237            58    1000300A990    Credit Card         NaN   
4     0b100001238            39    1000300A991      Home Loan         NaN   

   Unnamed: 5  Unnamed: 6  Unnamed: 7  
0         NaN         NaN         NaN  
1         NaN         NaN         NaN  
2         NaN         NaN         NaN  
3         NaN         NaN         NaN  
4         NaN         NaN         NaN

In [14]:
df["CUSTOMER_AGE"].count()

64

In [23]:
df_bin = pd.DataFrame( data ={"Bin":["18-29","30-44", "45-59", "60+"]})
df_bin.head()

Bin
0  18-29
1  30-44
2  45-59
3    60+

In [33]:
def impute_age_bin(cols):
    bin_18 = bin_30= bin_45= bin_60= 0
    if cols >= 18 and cols<30:
        bin_18 += bin_18
    elif cols >= 30 and cols<45:
        bin_30 += bin_30
    elif cols >= 45 and cols<60:
        bin_45 += bin_45
    else:
        bin_60 += bin_60
    
    return pd.DataFrame({"percent": map(lambda x: x*100/df["CUSTOMER_AGE"].count(),[bin_18,bin_30,bin_45,bin_60])})
    
    

In [39]:
df_bin["Percentage"] = df["CUSTOMER_AGE"].apply(impute_age_bin,axis=1)

TypeError: impute_age_bin() got an unexpected keyword argument 'axis'